crates/forge_analyzer/src/checkers.rs

```diff
@@ -886,6 +892,14 @@ impl SecretVuln {
             file,
         }
     }
+
+    fn from_manifest(location: String, field_name: String) -> Self {
```

Comment:
We should probably make this an `enum` to create a different `Vulnerability` in `into_vuln()` depending on whether it's an OAuth provider secret or regular hardcoded secret. That way the recommendation can be different, since the correct approach is to configure the `client_secret` and use runtime templates for OAuth provider hardcoded secrets.